packages/datadog-plugin-graphql/src/index.js

```diff
@@ -352,7 +355,7 @@ function finishResolvers (contextValue) {
 
 function updateField (field, error) {
   // TODO: update this to also work with no-op spans without a hack
-  field.finishTime = field.span._getTime ? field.span._getTime() : 0
+  field.finishTime = field.finishTime || (field.span._getTime ? field.span._getTime() : 0)
```

Comment:
This will cause array resolvers to end up with incorrect timing. Unfortunately, as you pointed out in the PR description, it's not easy to test timings, so this may not currently be tested properly, but it's the function purpose.

What specific problem was addressed by this change? Could it be done in another way?

Reply:
It's been a while since I wrote the underlying code, so I don't exactly remember what was going on, but this is what the traces look like if I remove it:

![Screen Shot 2021-08-23 at 7 02 21 PM](https://user-images.githubusercontent.com/1102667/130530220-c53657db-3557-4f1e-88f9-8d5cd63e9280.png)

IIRC, `updateField()` is called more than once for many fields (I _think_ as an unintentional side-effect of the delegated/wrapped/repeated `execute` calls made by graphql-tools).  For some reason, the tracer is looping through and _re_-finishing all of the spans and overwriting the correct values for `finishTime`.

Reply:
Calling `finishResolvers` multiple times would have no effect, since the `finish` function on a span becomes a no-op once the span has been finished once. Not sure however what else could be updating the finish time. Do you know if the multiple `execute` calls could somehow be re-running the resolvers?

Reply:
> However, when you do anything in graphql-tools that invokes delegateToSchema, execute effectively gets called (at least) twice for the entire query. Once to run the actual queries, and once more to transform and reassemble the results (via defaultMergedResolver).

If it executes twice, would it make sense to actually see the 2 executions? Since right now multiple executions of the same field are merged together because of arrays, maybe it makes sense that the 2 `parent` spans are merged together with both timings combined? I don't know Apollo well enough to know if that makes sense or not, but let me know what you think.

Reply:
> Any advice on how you'd handle this (and any ideas for why the defaultCreateProxyingResolver patch might not be working) would be appreciated!

I'm not sure, at a glance it looks good to me. The `@graphql-tools/delegate` instrumentation works properly?